### PR TITLE
[#1927] fix(dashboard): Fix NPE issues in dashboard

### PR DESCRIPTION
**Pull request by yl09099**
### What changes were proposed in this pull request?

1. Modify NPE bugs;
2. Modify the Vue3's CompositionAPI.
3. Handling the request failed.
4. Optimize page display.

image
image

### Why are the changes needed?

Fix: #1954 

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

UT.

### CODE REVIEW

**Comment by yl09099** [comment]
@jerqi @xianjingfeng @rickyma @kuszz @maobaolong PATL

File: conf/local_dev/dashboard.conf [hunk 1]
```diff
@@ -17,4 +17,4 @@
 
 rss.dashboard.http.port 19997
 # change to the actual coordinator web addresses
-coordinator.web.address http://localhost:19998,http://localhost:19998
+coordinator.web.address http://127.0.0.1:19998,http://localhost:19999
```
Comment (maobaolong) [inline conf/local_dev/dashboard.conf:20]: You can move this conf into a place out of git version control?

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java [hunk 2]
```diff
@@ -22,16 +22,22 @@
...
+import org.eclipse.jetty.http.HttpStatus;
 import org.eclipse.jetty.proxy.ProxyServlet;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class WebProxyServlet extends ProxyServlet {
 
   private static final Logger LOG = LoggerFactory.getLogger(WebProxyServlet.class);
+  /** The key of the request header. */
+  private static final String TARGETADDRESS = "targetAddress";
```
Comment (maobaolong) [inline dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:34]: TARGETADDRESS -> `HEADER_TARGET_ADDRESS ` would be better

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java [hunk 3]
```diff
@@ -22,16 +22,22 @@
...
 
+  private static final String REQUESTSERVERTYPE = "requestServerType";
```
Comment (maobaolong) [inline dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:36]: The same `REQUESTSERVERTYPE ` -> `HEADER_REQUEST_SERVER_TYPE`

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java [hunk 4]
```diff
@@ -22,16 +22,22 @@
...
+  /** The value of the request header. */
+  private static final String COORDINATOR = "coordinator";
```
Comment (maobaolong) [inline dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:38]: REQUEST_SERVER_TYPE_COORDINATOR

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java [hunk 5]
```diff
@@ -22,16 +22,22 @@
...
+
+  private static final String SERVER = "server";
```
Comment (maobaolong) [inline dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:40]: REQUEST_SERVER_TYPE_SERVER

**Review by maobaolong — CHANGES_REQUESTED** [review]
@yl09099 This looks pretty good like than before! Left some named issue.

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java (hunk 5 above)
Comment (yl09099) [reply dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java]: > REQUEST_SERVER_TYPE_SERVER

done.

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java [hunk 6]
```diff
@@ -46,47 +52,56 @@ protected String rewriteTarget(HttpServletRequest clientRequest) {
...
 
-  @Override
-  protected void onProxyRewriteFailed(
-      HttpServletRequest clientRequest, HttpServletResponse clientResponse) {}
-
+  /**
+   * If the proxy address fails to be requested, 403 is returned and the front-end handles the
+   * exception.
+   *
+   * @param clientRequest
```
Comment (maobaolong) [inline dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:81]: These java docs seems generated by development tools, but uncompleted?

**Review by maobaolong — COMMENTED** [review]
Looks good overall, look forward to merge this as soon as possible.

Another feature request related, do you think it is easy to show or download server log coordinator log in dashboard?

File: dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java (hunk 6 above)
Comment (yl09099) [reply dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:81]: > These java docs seems generated by development tools, but uncompleted?

I made it easy, but it was done.

Comment (maobaolong) [reply dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java:81]: Its ok.

**Comment by maobaolong** [comment]
@yl09099 Why the color of each button are different between coordinator page and server node page

**Comment by yl09099** [comment]
> @yl09099 Why the color of each button are different between coordinator page and server node page

OK,Let me change the color.

**Comment by yl09099** [comment]
> @yl09099 Why the color of each button are different between coordinator page and server node page

Modified, color consistent.

**Review by maobaolong — APPROVED** [review]
@yl09099 It LGTM overall, but i'm not familiar with vue and js. Thanks for this contribution!

**Review by jerqi — APPROVED** [review]

**Comment by rickyma** [comment]
Merged to master. Thanks for your contribution. @yl09099 Thanks for your reviews. @maobaolong @jerqi

**Pull request merged**